inc/umock_c/umock_c_internal.h

```diff
@@ -1198,9 +1198,9 @@ typedef struct MOCK_CALL_METADATA_TAG
         },) \
         return result; \
     } \
-    MU_IF(IS_NOT_VOID(return_type),static return_type MU_C2(mock_call_default_result_,name); \
-    static return_type MU_C2(mock_call_fail_result_,name); \
-    IMPLEMENT_SET_RETURN_FUNCTION(return_type, name, __VA_ARGS__) \
+    MU_IF(IS_NOT_VOID(return_type),static return_type MU_C2(mock_call_default_result_,name) MU_IF(MU_COUNT_ARG(success_return_value), = success_return_value,);,) \
```

Comment:
Can you move the MU_IF(IS_NOT_VOID) such that there's only one check here?  I'm mostly concerned RE performance given preprocessors have problems on macros this complex so not sure how this impacts that.

Reply:
Sure, done!

---
In reply to: [280245414](https://github.com/Azure/umock-c/pull/72#discussion_r280245414) [](ancestors = 280245414)